AutoRest/Generators/CSharp/CSharp/TemplateModels/ModelTemplateModel.cs

```diff
@@ -187,6 +198,49 @@ private static string CreateSignature(IEnumerable<ConstructorParameterModel> par
                 return string.Join(", ", declarations);
             }
 
+            private static IEnumerable<string> CreateSignatureDocumentation(IEnumerable<ConstructorParameterModel> parameters)
+            {
+                var declarations = new List<string>();
+                IEnumerable<ConstructorParameterModel> parametersWithDocumentation =
+                    parameters.Where(p => !(string.IsNullOrEmpty(p.UnderlyingProperty.Summary) && string.IsNullOrEmpty(p.UnderlyingProperty.Documentation)));
+
+                foreach (var property in parametersWithDocumentation.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
+                {
+                    string documentation = string.Empty;
+                    string formatString = "<param name=\"{0}\">{1}</param>";
+
+                    if (string.IsNullOrEmpty(property.Documentation))
+                    {
+                        documentation = string.Format(
+                            CultureInfo.InvariantCulture,
+                            formatString,
+                            char.ToLower(property.Name[0]) + property.Name.Substring(1),
+                            string.Empty);
+                    }
+                    else
+                    {
+                        documentation = string.Format(
+                            CultureInfo.InvariantCulture,
+                            formatString,
+                            char.ToLower(property.Name[0]) + property.Name.Substring(1),
+                            property.Documentation);
+                    }
```

Comment:
This whole `if` `else` is actually identical except for the last parameter -- instead of doing `if` `else` like this, just do:
```
var documentationInnerText = property.Summary ?? property.Documentation ?? string.Empty
documentation = string.Format(
                            CultureInfo.InvariantCulture,
                            formatString,
                            char.ToLower(property.Name[0]) + property.Name.Substring(1),
                            documentationInnerText);
```